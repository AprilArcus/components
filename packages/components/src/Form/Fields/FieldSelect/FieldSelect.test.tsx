/*

 MIT License

 Copyright (c) 2019 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import { fireEvent } from '@testing-library/react'
import React from 'react'

import { renderWithTheme } from '@looker/components-test-utils'

// import { Label } from '../../Label/Label'
import { FieldSelect } from './FieldSelect'

describe('<FieldSelect/>', () => {
  test('value', () => {
    const { getAllByLabelText } = renderWithTheme(
      <FieldSelect
        label="Thumbs Up"
        name="thumbsUp"
        id="thumbs-up"
        value="Foo bar"
        options={[{ data: 'foobar', value: 'Foo bar' }]}
      />
    )

    const input = getAllByLabelText('Thumbs Up')[1]
    expect(input).toHaveValue('Foo bar')
  })

  test('input onChange handler', () => {
    const handleChange = jest.fn()

    const { getByLabelText } = renderWithTheme(
      <FieldSelect
        label="Thumbs Up"
        name="thumbsUp"
        id="thumbs-up"
        value="foobar"
        required
        inputProps={{ onChange: handleChange }}
      />
    )

    const input = getByLabelText('Thumbs Up')
    fireEvent.change(input, { target: { value: '' } })
    expect(handleChange).toHaveBeenCalledTimes(1)
  })

  test('required', () => {
    const { getByLabelText } = renderWithTheme(
      <FieldSelect label="Thumbs Up" name="thumbsUp" id="thumbs-up" required />
    )
    const input = getByLabelText('Thumbs Up')
    expect(input).toHaveAttribute('required')
  })
})
