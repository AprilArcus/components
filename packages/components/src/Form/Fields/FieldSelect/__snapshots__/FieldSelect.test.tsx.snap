// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A FieldSelect 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #262D33;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.875rem;
  display: inline-block;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: solid 1px;
  border-color: #C1C6CC;
  border-radius: 0.25rem;
  width: 100%;
  height: 28px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-size: 0.875rem;
}

.c5 {
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTcuNDEgOEwxMiAxMi41OEwxNi41OSA4TDE4IDkuNDFMMTIgMTUuNDFMNiA5LjQxTDcuNDEgOFoiIGZpbGw9IiM3MDc3ODEiLz4KPC9zdmc+');
  background-repeat: no-repeat,repeat;
  background-position: right .25rem center,0 0;
  background-size: 1rem,100%;
  padding-right: calc(2 * .25rem + 1rem);
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="palette.charcoal800"
    fontSize="small"
    fontWeight="semiBold"
    htmlFor="thumbs-up"
    id="label-thumbs-up"
  >
    üëç
  </label>
  <div
    className="c0"
  >
    <div
      className="c2"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="listbox-thumbs-up"
        className="c3 "
        id="thumbs-up"
        name="thumbsUp"
        role="select"
      >
        <input
          className="c4 c5"
          id="listbox-thumbs-up"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`A FieldSelect with an error validation aligned to the left 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #262D33;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.875rem;
  display: inline-block;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: solid 1px;
  border-color: #C1C6CC;
  border-radius: 0.25rem;
  width: 100%;
  height: 28px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-size: 0.875rem;
}

.c6 {
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTcuNDEgOEwxMiAxMi41OEwxNi41OSA4TDE4IDkuNDFMMTIgMTUuNDFMNiA5LjQxTDcuNDEgOFoiIGZpbGw9IiM3MDc3ODEiLz4KPC9zdmc+');
  background-repeat: no-repeat,repeat;
  background-position: right .25rem center,0 0;
  background-size: 1rem,100%;
  padding-right: calc(2 * .25rem + 1rem);
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c7 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #990F14;
  margin-right: 0.5rem;
  margin-top: 0.5rem;
  font-size: 0.75rem;
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="palette.charcoal800"
    fontSize="small"
    fontWeight="semiBold"
    htmlFor="thumbs-up"
    id="label-thumbs-up"
  >
    üëç
  </label>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="listbox-thumbs-up"
        className="c4 "
        id="thumbs-up"
        name="thumbsUp"
        role="select"
        validationType="error"
      >
        <input
          className="c5 c6"
          id="listbox-thumbs-up"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="c7"
      fontSize="xsmall"
      type="error"
    >
      This is an error
    </div>
  </div>
</div>
`;

exports[`A FieldSelect with an error validation aligned to the right 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #262D33;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.875rem;
  display: inline-block;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: solid 1px;
  border-color: #C1C6CC;
  border-radius: 0.25rem;
  width: 100%;
  height: 28px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-size: 0.875rem;
}

.c6 {
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTcuNDEgOEwxMiAxMi41OEwxNi41OSA4TDE4IDkuNDFMMTIgMTUuNDFMNiA5LjQxTDcuNDEgOFoiIGZpbGw9IiM3MDc3ODEiLz4KPC9zdmc+');
  background-repeat: no-repeat,repeat;
  background-position: right .25rem center,0 0;
  background-size: 1rem,100%;
  padding-right: calc(2 * .25rem + 1rem);
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c7 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #990F14;
  margin-right: 0.5rem;
  margin-top: 0.5rem;
  margin-left: 0.5rem;
  font-size: 0.75rem;
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="palette.charcoal800"
    fontSize="small"
    fontWeight="semiBold"
    htmlFor="thumbs-up"
    id="label-thumbs-up"
  >
    üëç
  </label>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="listbox-thumbs-up"
        className="c4 "
        id="thumbs-up"
        name="thumbsUp"
        role="select"
        validationType="error"
      >
        <input
          className="c5 c6"
          id="listbox-thumbs-up"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="c7"
      fontSize="xsmall"
      type="error"
    >
      This is an error
    </div>
  </div>
</div>
`;

exports[`A required FieldSelect 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #262D33;
  margin-right: 0.5rem;
  font-weight: 600;
  font-size: 0.875rem;
  display: inline-block;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: solid 1px;
  border-color: #C1C6CC;
  border-radius: 0.25rem;
  width: 100%;
  height: 28px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-size: 0.875rem;
}

.c6 {
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTcuNDEgOEwxMiAxMi41OEwxNi41OSA4TDE4IDkuNDFMMTIgMTUuNDFMNiA5LjQxTDcuNDEgOFoiIGZpbGw9IiM3MDc3ODEiLz4KPC9zdmc+');
  background-repeat: no-repeat,repeat;
  background-position: right .25rem center,0 0;
  background-size: 1rem,100%;
  padding-right: calc(2 * .25rem + 1rem);
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  color: #990F14;
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="palette.charcoal800"
    fontSize="small"
    fontWeight="semiBold"
    htmlFor="thumbs-up"
    id="label-thumbs-up"
  >
    üëç
    <span
      aria-hidden="true"
      className="c2"
    >
       
      *
    </span>
  </label>
  <div
    className="c0"
  >
    <div
      className="c3"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="listbox-thumbs-up"
        className="c4 "
        id="thumbs-up"
        name="thumbsUp"
        required={true}
        role="select"
      >
        <input
          className="c5 c6"
          id="listbox-thumbs-up"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldSelect supports labelWeight 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  color: #262D33;
  margin-right: 0.5rem;
  font-weight: 400;
  font-size: 0.875rem;
  display: inline-block;
}

.c4 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: solid 1px;
  border-color: #C1C6CC;
  border-radius: 0.25rem;
  width: 100%;
  height: 28px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  font-size: 0.875rem;
}

.c5 {
  background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTcuNDEgOEwxMiAxMi41OEwxNi41OSA4TDE4IDkuNDFMMTIgMTUuNDFMNiA5LjQxTDcuNDEgOFoiIGZpbGw9IiM3MDc3ODEiLz4KPC9zdmc+');
  background-repeat: no-repeat,repeat;
  background-position: right .25rem center,0 0;
  background-size: 1rem,100%;
  padding-right: calc(2 * .25rem + 1rem);
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="palette.charcoal800"
    fontSize="small"
    fontWeight="normal"
    htmlFor="thumbs-up"
    id="label-thumbs-up"
  >
    üëç
  </label>
  <div
    className="c0"
  >
    <div
      className="c2"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="listbox-thumbs-up"
        className="c3 "
        id="thumbs-up"
        name="thumbsUp"
        role="select"
      >
        <input
          className="c4 c5"
          id="listbox-thumbs-up"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
